src/Pos/Delegation/DB.hs

```diff
@@ -119,19 +118,25 @@ isIssuerByAddressHash = fmap isJust . getPskByIssuer . Right
 getDlgTransitive :: MonadDBPure m => Either PublicKey StakeholderId -> m (Maybe PublicKey)
 getDlgTransitive (toStakeholderId -> issuer) = gsGetBi (transDlgKey issuer)
 
--- | Retrieves last PSK in chain of delegation started by public key.
-getDlgTransPsk :: MonadDBPure m => Either PublicKey StakeholderId -> m (Maybe ProxySKHeavy)
+-- | Retrieves last PSK in chain of delegation started by public key
+-- and resolves the passed issuer to a public key. Doesn't check that
+-- user himself didn't delegate.
+getDlgTransPsk
+    :: MonadDBPure m
+    => Either PublicKey StakeholderId -> m (Maybe (PublicKey, ProxySKHeavy))
 getDlgTransPsk issuer = getDlgTransitive issuer >>= \case
     Nothing -> pure Nothing
     Just dPk -> do
-        psks <- filter (\psk -> pskDelegatePk psk == dPk) . HM.elems <$>
-                getPskChain issuer
-        when (length psks > 1) $ throwM $ DBMalformed $
-            sformat ("getDlgTransPk: found several psks with a same dlg: "%listJson)
-                    psks
+        chain <- HM.elems <$> getPskChain issuer
+        let finalPsk = find (\psk -> pskDelegatePk psk == dPk) chain
+        let issuer' = toStakeholderId issuer
+        let iPk = pskIssuerPk <$>
+                  find (\psk -> addressHash (pskIssuerPk psk) == issuer') chain
```

Comment:
Yes, If somebody passes `Left pk`, we can just return `pk` but for the sake of compactnes we will still retrieve this `pk` from the beginning of the chain.